cmake/public/LoadHIP.cmake

```diff
@@ -1,246 +1,273 @@
+      set(file "${PROJECT_BINARY_DIR}/hipblaslt_test_data_type.cc")
+      file(WRITE ${file} ""
+	"#include <hipblaslt/hipblaslt.h>\n"
+	"int main() {\n"
+	"    hipblasltDatatype_t bar = HIPBLASLT_R_16F;\n"
+	"    return 0;\n"
+	"}\n"
       )
 
-    try_compile(hipblaslt_compile_result ${PROJECT_RANDOM_BINARY_DIR} ${file}
-      CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${ROCM_INCLUDE_DIRS}"
-      COMPILE_DEFINITIONS -D__HIP_PLATFORM_AMD__ -D__HIP_PLATFORM_HCC__
-      OUTPUT_VARIABLE hipblaslt_compile_output)
+      try_compile(hipblaslt_compile_result ${PROJECT_RANDOM_BINARY_DIR} ${file}
+	CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${ROCM_INCLUDE_DIRS}"
+	COMPILE_DEFINITIONS -D__HIP_PLATFORM_AMD__ -D__HIP_PLATFORM_HCC__
+	OUTPUT_VARIABLE hipblaslt_compile_output)
+
+      if(hipblaslt_compile_result)
+	set(HIPBLASLT_CUSTOM_DATA_TYPE ON)
+	#message("hipblaslt is using custom data type: ${hipblaslt_compile_output}")
+	message("hipblaslt is using custom data type")
+      else()
+	set(HIPBLASLT_CUSTOM_DATA_TYPE OFF)
+	#message("hipblaslt is NOT using custom data type: ${hipblaslt_compile_output}")
+	message("hipblaslt is NOT using custom data type")
+      endif()
+
+      # check whether hipblaslt is using its own compute type
+      set(file "${PROJECT_BINARY_DIR}/hipblaslt_test_compute_type.cc")
+      file(WRITE ${file} ""
+	"#include <hipblaslt/hipblaslt.h>\n"
+	"int main() {\n"
+	"    hipblasLtComputeType_t baz = HIPBLASLT_COMPUTE_F32;\n"
+	"    return 0;\n"
+	"}\n"
+      )
 
-    if(hipblaslt_compile_result)
-      set(HIPBLASLT_CUSTOM_COMPUTE_TYPE ON)
-      #message("hipblaslt is using custom compute type: ${hipblaslt_compile_output}")
-      message("hipblaslt is using custom compute type")
-    else()
-      set(HIPBLASLT_CUSTOM_COMPUTE_TYPE OFF)
-      #message("hipblaslt is NOT using custom compute type: ${hipblaslt_compile_output}")
-      message("hipblaslt is NOT using custom compute type")
+      try_compile(hipblaslt_compile_result ${PROJECT_RANDOM_BINARY_DIR} ${file}
+	CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${ROCM_INCLUDE_DIRS}"
+	COMPILE_DEFINITIONS -D__HIP_PLATFORM_AMD__ -D__HIP_PLATFORM_HCC__
+	OUTPUT_VARIABLE hipblaslt_compile_output)
+
+      if(hipblaslt_compile_result)
+	set(HIPBLASLT_CUSTOM_COMPUTE_TYPE ON)
+	#message("hipblaslt is using custom compute type: ${hipblaslt_compile_output}")
+	message("hipblaslt is using custom compute type")
+      else()
+	set(HIPBLASLT_CUSTOM_COMPUTE_TYPE OFF)
+	#message("hipblaslt is NOT using custom compute type: ${hipblaslt_compile_output}")
+	message("hipblaslt is NOT using custom compute type")
+      endif()
     endif()
+
   endif()
+else()
+  find_package(ROCM CONFIG REQUIRED)
+  find_package(HIP MODULE REQUIRED)
+  message("HIP Version: ${HIP_VERSION}")
+  math(EXPR TORCH_HIP_VERSION "(${HIP_VERSION_MAJOR} * 100) + ${HIP_VERSION_MINOR}")
+  math(EXPR ROCM_VERSION_DEV_INT "(${HIP_VERSION_MAJOR} * 1000) + ${HIP_VERSION_MINOR} * 100")
+  
+  find_package(hsa-runtime64 REQUIRED)
```

Comment:
Instead of introducing a duplicated list of packages (which will increase maintenance burden), can we use the `foreach` CMake command to loop through a package list, and use the same package list for both system and non-system ROCm cases?